src/server/status_server/lite.rs

```diff
@@ -0,0 +1,224 @@
+use super::{make_response, tls_incoming, StatusServer};
+use crate::server::Result;
+
+/// Svc is a type alias that help us to call static methods in the full status
+/// server.
+type Svc = StatusServer<()>;
+
+/// Server manages how we accept the incoming requests and how we handle them.
+///
+/// After creating and configurating this, you may use [`start`] to start
+/// serving and detach this. You can control the server then by the [`Handle`].
+pub struct Server {
+    security_config: Arc<SecurityConfig>,
+}
+
+/// Handle is the controller for the sever.
+pub struct Handle {
+    addr: SocketAddr,
+}
+
+impl Handle {
+    /// Return the bound address of the server.
+    pub fn address(&self) -> &SocketAddr {
+        &self.addr
+    }
+}
+
+impl Server {
+    pub fn new(sec: Arc<SecurityConfig>) -> Self {
+        Server {
+            security_config: sec,
+        }
+    }
+
+    /// Start the server.
+    ///
+    /// # Panic
+    ///
+    /// This must be run in a tokio context. Or this will panic.
+    pub fn start(self, status_addr: &str) -> Result<Handle> {
+        let addr = SocketAddr::from_str(status_addr)?;
+
+        let incoming = AddrIncoming::bind(&addr)?;
+        let hnd = Handle {
+            addr: incoming.local_addr(),
+        };
+        if !self.security_config.cert_path.is_empty()
+            && !self.security_config.key_path.is_empty()
+            && !self.security_config.ca_path.is_empty()
+        {
+            let tls_incoming = tls_incoming(self.security_config.clone(), incoming)?;
+            let server = HyperSrv::builder(tls_incoming);
+            self.start_serve(server);
+        } else {
+            let server = HyperSrv::builder(incoming);
+            self.start_serve(server);
+        }
+        Ok(hnd)
+    }
+
+    fn start_serve<I, C>(self, builder: hyper::server::Builder<I>)
+    where
+        I: Accept<Conn = C, Error = std::io::Error> + Send + 'static,
+        I::Error: Into<Box<dyn StdError + Send + Sync>>,
+        I::Conn: AsyncRead + AsyncWrite + Unpin + Send + 'static,
+        C: super::ServerConnection,
+    {
+        let mut svc = LiteService;
+
+        let server = builder.serve(super::make_service_fn(move |conn: &C| {
+            let client_cert = conn.get_x509();
+            let security = self.security_config.clone();
+            std::future::ready(hyper::Result::Ok(service_fn(move |req| {
+                let client_cert = client_cert.clone();
+                let security = security.clone();
+                async move {
+                    svc.call(RequestCtx {
+                        req,
+                        client_cert,
+                        security,
+                    })
+                    .await
+                }
+            })))
+        }));
+
+        let svc =
+            server.map_err(|err| warn!("status server lite encountered error"; "err" => %err));
+        tokio::spawn(svc);
+    }
+}
+
+#[derive(Copy, Clone)]
+pub struct LiteService;
```

Comment:
quick question, what does Lite mean here?

Reply:
"light wighted", it truncates most storage-engine related endpoints and only keeps the basic state exporters (mertics, cpu / heap profiling, etc..).

Reply:
gotcha, if that's case I would recommend to name it `StateExportOnlySerivce`or something like that, or maybe add a comment to describe what it does?

Reply:
The comment was added in the module level. See the top of the file.